lib/omniauth/strategies/openid_connect.rb

```diff
@@ -207,7 +211,8 @@ def access_token
       end
 
       def decode_id_token(id_token)
-        ::OpenIDConnect::ResponseObject::IdToken.decode(id_token, public_key)
+        # NOTE: we might want to cache the decoded token so it can be used to set basic user info
+        @decoded_id_token ||= ::OpenIDConnect::ResponseObject::IdToken.decode(id_token, public_key)
```

Comment:
Please, do not memoize result of instance method if the result depends on the passed argument

Reply:
Nice catch, I'll use Hash instead. We might want to rename the method as well, since it returns the decodeD token, it can be renamed to `decoded_id_token`. Also, since it takes the parameter, we can pluralize it : `decoded_id_tokens()`

Reply:
No, please, don't rename this method. It works just fine now and does what it supposed to to do - decode passed token.

Reply:
Right, I added another method that uses caching with Hash and calls the original method to do the job